source/game/system/map/MapdataCheckPoint.cc

```diff
@@ -95,15 +95,120 @@ void MapdataCheckPoint::initCheckpointLinks(MapdataCheckPointAccessor &accessor,
     }
 }
 
+/// @addr{0x80510D7C}
+/// @see MapdataCheckPoint::checkSectorAndDistanceRatio
+/// @rename
+MapdataCheckPoint::SectorOccupancy MapdataCheckPoint::checkSectorAndDistanceRatio(
+        const EGG::Vector3f &pos, f32 &completion) const {
+    EGG::Vector2f p1 = m_right;
+    p1.y = pos.z - p1.y;
+    p1.x = pos.x - p1.x;
+
+    for (auto &next : m_nextPoints) {
+        EGG::Vector2f p0 = next.checkpoint->m_left;
+        p0.y = pos.z - p0.y;
+        p0.x = pos.x - p0.x;
+        MapdataCheckPoint::SectorOccupancy result =
+                checkSectorAndDistanceRatio(next, p0, p1, completion);
+
+        if (result != SectorOccupancy::OutsideSector) {
+            return result;
+        }
+    }
+
+    return SectorOccupancy::OutsideSector;
```

Comment:
This is an inaccuracy. We are only allowed to return early if we're inside the sector. Otherwise, we take the best sector occupancy result.

The solution is to use a boolean named `betweenSides`, and changing the logic to behave like this:
```suggestion
        if (result == SectorOccupancy::InsideSector) {
            return SectorOccupancy::InsideSector;
        } else if (result == SectorOccupancy::BetweenSides) {
            betweenSides = true;
        }
    }

    return betweenSides ? SectorOccupancy::BetweenSides : SectorOccupancy::OutsideSector;
```